dbcrossbarlib/src/drivers/redshift/mod.rs

```diff
@@ -146,10 +147,11 @@ impl LocatorStatic for RedshiftLocator {
 /// Given a `DriverArgs` structure, convert it into Redshift credentials SQL.
 pub(crate) fn credentials_sql(args: &DriverArguments) -> Result<String> {
     let mut out = vec![];
-    for (k, v) in args.iter() {
+    let map = args.deserialize::<HashMap<String, String>>()?;
+    for (k, v) in &map {
         lazy_static! {
             static ref KEY_RE: Regex =
-                Regex::new("^[-_A-Za-z0-9]+$").expect("invalid regex in source code");
+                Regex::new("^[_A-Za-z0-9]+$").expect("invalid regex in source code");
```

Comment:
heh love is in the details

Reply:
I'm not actually sure whether `-` is legal with RedShift, but I can't find any obvious examples, and it won't make it past the new arg parser anyway. So I'm going to take it out and see what happens.